src/operator/tensor/matrix_op-inl.h

```diff
@@ -675,20 +675,31 @@ inline void GetIndexRange(const mxnet::TShape& dshape,
         b = len - 1;
       }
       CHECK_LT(b, len) << "slicing with begin[" << i << "]="
-                       << b << " exceends limit of " << len;
+                       << b << " exceeds limit of " << len;
 
       if (param_end[i].has_value()) {
         e = param_end[i].value();
-        if (e < 0) {
-          e += len;
-          CHECK_GE(e, 0) << "slicing with end[" << i << "]="
-                         << e - len << " exceeds limit of " << len;
+        if (s > 0) {
+          if (e < 0) {
+            e += len;
+            CHECK_GE(e, 0) << "slicing with end[" << i << "]="
+                           << e - len << " exceeds limit of " << len;
+          }
+        } else {
+          if (e < -1) {
+            e += len;
+            CHECK_GE(e, -1) << "slicing with end[" << i << "]="
+                           << e - len << " exceeds limit of " << len;
```

Comment:
Can we give a more informative error message than just `len`? e.g. "the shape size `len` in dimension `i`"?